src/praisonai-agents/tests/unit/session/test_session_docs_examples.py

```
@@ -33,20 +41,180 @@ def isolated_home(tmp_path, monkeypatch):
+
+    monkeypatch.setattr(Agent, "start", lambda self, *a, **k: "stub", raising=False)
+    monkeypatch.setattr(Agent, "chat", lambda self, *a, **k: "stub", raising=False)
+
+    if _WRAPPER_BACKED.search(code):
+        pytest.importorskip(
+            "praisonai",
+            reason="example uses a db backend from the optional praisonai wrapper",
+        )
+
+    try:
+        exec(compile(code, f"<{_block_id(block)}>", "exec"),
```

Ruff (0.16.1)

[error] Use of `exec` detected

(S102)